src/logger.jl

```diff
@@ -100,42 +105,58 @@ Global `Logger` being used to record the events. Can be changed using [`setlogge
 """
 const LOGGER = Ref{Logger}()
 
-
 ############################################################################
 #                           To Plot DAG
 ############################################################################
+Base.isless(t1::TaskLog,t2::TaskLog) = isless(t1.tag,t2.tag)
 
+function topological_sort(l::Logger)
+    tlogs = Iterators.flatten(l.tasklogs) |> collect
+    sort!(tlogs)
+end
 
 """
     criticalpath(logger) --> path
+
 Finds the critical path of the logger's DAG
 """
 function criticalpath(logger=getlogger())
+    # check that logger contains tasks with contiguous tags (not necessarily
+    # starting at 1)
+    tasklogs = topological_sort(logger)
+    nbnodes  = length(tasklogs)
+    for i in 2:nbnodes
+        tag(tasklogs[i]) == tag(tasklogs[i-1]) + 1 || error("tasks not contiguous")
+    end
+    # HACK: because the current algorithm assumes the tasks start at one, apply
+    # a shift here and at the end
+    shift = tag(first(tasklogs)) - 1
```

Comment:
This is a hacky way to account for the fact that the critical path algorithm makes some strong (implicit) assumptions about the tags associated with the tags. In particular, it assumes that the tags start at one, which in turn means that the `resetlogger!` had to reset the task counter. Reseting the task counter should be avoided as it can mess up with the precedence notion of the tasks.

This is not really a fix for #14, but rather a quick-and-dirty workaround. I still think the graph algorithms should be made more robust in the future.